src/libstore/build/derivation-goal.cc

```diff
@@ -1267,6 +1270,11 @@ void DerivationGoal::checkPathValidity()
             }
         }
     }
+    // If we requested all the outputs via the empty set, we are always fine.
+    // If we requested specific elements, the loop above removes all the valid
+    // ones, so any that are left must be invalid.
+    if (!wantedOutputsLeft.empty())
+        throw UsageError("some wanted outputs are not provided by the derivation: %s", concatStringsSep(", ", wantedOutputsLeft));
```

Comment:
This shouldn't be a `UsageError` but a regular error. `UsageError` is for command-line arguments that are never correct. 

We should probably also show the `drvPath` here.

Reply:
Fixed both.